libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/Span.java

```diff
@@ -62,6 +62,13 @@ public interface Span {
      */
     void addAttribute(String key, Boolean value);
 
+    /**
+     * Sets status of the span
+     *
+     * @param statusCode status of the span
+     */
+    void setStatus(StatusCode statusCode);
```

Comment:
I don't think we need the setStatus and StatusCode: the typical way the errors are reported in Java are through exceptions, so we need `error` / `setError` to propagate that.

```
try (Scope s = tracer.startSpan(...)) {
  ...
} catch (Exception ex) {
  s.error(ex);
}
```

Reply:
Sounds reasonable as we have mainly two statuses error and non error. 
So in our Scope implementation we save it as status in Otel ?
```
public void setStatus(Exception e) {
        delegateSpan.setStatus(StatusCode.ERROR, e.getMessage());
    }
```

Reply:
Addressed in latest commit